### PR TITLE
Move MySQL state tests to pytest

### PR DESCRIPTION
**Pull request by krionbsd**
### What does this PR do?
Move MySQL state tests to pytest

### CODE REVIEW

File: tests/pytests/unit/states/test_mysql_database.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {mysql_database: {}}
+
+
+def test_present():
+    """
+    Test to ensure that the named user is present with
```
Comment (Ch3LL) [inline tests/pytests/unit/states/test_mysql_database.py:16]:
```suggestion
    Test to ensure that the named database is present with
```

File: tests/pytests/unit/states/test_mysql_database.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            with patch.object(mysql_database, "_get_mysql_error", mock_no_err):
+                ret.update({"changes": {}})
+                comt = "Failed to create database {}".format(dbname)
+                ret.update({"comment": comt, "result": False})
+                assert mysql_database.present(dbname) == ret
+
+
+def test_absent():
+    """
+    Test to ensure that the named user is absent.
```
Comment (Ch3LL) [inline tests/pytests/unit/states/test_mysql_database.py:125]:
```suggestion
    Test to ensure that the named database is absent.
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/states/test_mysql_query.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    import MySQLdb
+
+    HAS_PyMYSQL = True
+except ImportError:
+    HAS_PyMYSQL = False
+
+skip_on_missing_mysql = pytest.mark.skipif(
+    HAS_MYSQL is False and HAS_PyMYSQL is False,
+    reason="Install MySQL bindings before running MySQL unit tests.",
+)
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_mysql_query.py:39]:
```suggestion
MySQLdb = pytest.importorskip("MySQLdb")
pymysql = pytest.importorskip("pymysql")

pymysql.install_as_MySQLdb()
```

File: tests/pytests/unit/states/test_mysql_query.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                == ret
+            )
+
+        with patch.dict(mysql_query.__opts__, {"test": False}):
+            ret.update({"comment": "salt", "changes": {"query": "Executed"}})
+            assert mysql_query.run(name, database, query) == ret
+
+
+@skip_on_missing_mysql
+def test_run_multiple_statements():
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_mysql_query.py:171]:
```suggestion
def test_run_multiple_statements():
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
Additionally, let's create the mysql tests under it's own directory:
```
tests/pytests/unit/states/mysql/
├── __init__.py
├── test_database.py
├── test_grants.py
├── test_query.py
└── test_user.py
```

File: tests/pytests/unit/states/test_mysql_query.py (hunk 3 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_mysql_query.py]: Fixed

File: tests/filename_map.yml [hunk 5]
```diff
@@ -57,9 +57,9 @@ salt/modules/influxdb08mod.py:
   - pytests.unit.states.test_influxdb08_user
 
 salt/modules/mysql.py:
-  - unit.states.test_mysql_user
-  - unit.states.test_mysql_query
-  - unit.states.test_mysql_grants
+  - pytests.unit.states.mysql.test_mysql_user
+  - pytests.unit.states.mysql.test_mysql_query
+  - pytests.unit.states.mysql.test_mysql_grants
```
Comment (s0undt3ch) [inline tests/filename_map.yml:62]:
```suggestion
  - pytests.unit.states.mysql.test_user
  - pytests.unit.states.mysql.test_query
  - pytests.unit.states.mysql.test_grants
```
And also rename the test modules, there's no need to have `mysql` in the path twice.

File: tests/pytests/unit/states/mysql/test_mysql_query.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+import pytest
+import salt.modules.mysql as mysql_mod
+import salt.states.mysql_query as mysql_query
+from tests.support.mock import MagicMock, patch
+
+log = logging.getLogger(__name__)
+
+HAS_PyMYSQL = False
+HAS_MYSQL = False
+
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/mysql/test_mysql_query.py:17]:
```suggestion
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/filename_map.yml (hunk 5 above)
Comment (krionbsd) [reply tests/filename_map.yml]: Fixed

File: tests/pytests/unit/states/mysql/test_mysql_query.py (hunk 6 above)
Comment (krionbsd) [reply tests/pytests/unit/states/mysql/test_mysql_query.py]: Nice catch, I missed this one

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**